### PR TITLE
Increased the probability of generating the same hash more than once

### PR DESCRIPTION
**Pull request by Soupstraw**
# Description

This is supposed to make it easier to find a counterexample for IntersectMBO/formal-ledger-specifications#514

I managed to catch the bug with the modified generators by making it more likely to generate the same credential in multiple places. I also created an ImpTest that uses a manually constructed state to reproduce the bug.

# Checklist

- [ ] Commit sequence broadly makes sense and commits have useful messages
- [ ] New tests are added if needed and existing tests are updated
- [ ] All visible changes are prepended to the latest section of a `CHANGELOG.md` for the affected packages.
      **_New section is never added with the code changes._** (See [RELEASING.md](https://github.com/intersectmbo/cardano-ledger/blob/master/RELEASING.md#changelogmd))
- [ ] When applicable, versions are updated in `.cabal` and `CHANGELOG.md` files according to the
      [versioning process](https://github.com/intersectmbo/cardano-ledger/blob/master/RELEASING.md#versioning-process).
- [ ] The version bounds in `.cabal` files for all affected packages are updated.
      **_If you change the bounds in a cabal file, that package itself must have a version increase._** (See [RELEASING.md](https://github.com/intersectmbo/cardano-ledger/blob/master/RELEASING.md#versioning-process))
- [x] Code is formatted with [`fourmolu`](https://github.com/fourmolu/fourmolu) (use `scripts/fourmolize.sh`)
- [x] Cabal files are formatted (use `scripts/cabal-format.sh`)
- [x] [`hie.yaml`](https://github.com/intersectmbo/cardano-ledger/blob/master/hie.yaml) has been updated (use `scripts/gen-hie.sh`)
- [ ] Self-reviewed the diff

### CODE REVIEW

File: libs/cardano-ledger-conformance/src/Test/Cardano/Ledger/Conformance/ExecSpecRule/Conway/Base.hs [hunk 1]
```diff
@@ -267,12 +268,8 @@ ratifyEnvSpec ConwayRatifyExecContext {crecGovActionMap} =
...
           ( \eval ->
               pure $
-                Set.map
-                  CommitteeHotCredential
-                  ( Set.intersection
-                      (authorizedHotCommitteeCredentials (eval committeeState))
-                      ccVotes
-                  )
+                Set.map CommitteeHotCredential ccVotes
+                  `Set.intersection` foldr' Set.insert mempty (eval cs)
```
Comment (teodanciu) [inline libs/cardano-ledger-conformance/src/Test/Cardano/Ledger/Conformance/ExecSpecRule/Conway/Base.hs:315]: Out of curiosity, why this change?

File: libs/cardano-ledger-conformance/src/Test/Cardano/Ledger/Conformance/ExecSpecRule/Conway/Base.hs [hunk 2]
```diff
@@ -349,6 +347,18 @@ ratifyStateSpec _ RatifyEnv {..} =
     disableBootstrap pp = match pp $ \pp' ->
       match (sel @12 pp') $ \major _ ->
         assert $ not_ (major ==. lit (natVersion @9))
```
Comment (teodanciu) [inline libs/cardano-ledger-conformance/src/Test/Cardano/Ledger/Conformance/ExecSpecRule/Conway/Base.hs]: I think below 9 doesn't make sense either, no?  For most tests I set it to 10 in the environment spec,, via `satisfies pp pparamsSpec`, (along with some other constraints for pparams) -  maybe it would make sense to have this here as well?

File: eras/conway/impl/src/Cardano/Ledger/Conway/Rules/Ratify.hs [hunk 3]
```diff
@@ -20,6 +20,7 @@ module Cardano.Ledger.Conway.Rules.Ratify (
   spoAcceptedRatio,
   dRepAccepted,
   dRepAcceptedRatio,
+  acceptedByEveryone,
```
Comment (teodanciu) [inline eras/conway/impl/src/Cardano/Ledger/Conway/Rules/Ratify.hs:23]: If you wanted you could add to conway CHANGELOG
Added `showGovActionType`  and  `acceptedByEveryone`

**Review by teodanciu — APPROVED** [review]
The message of the first commit seems totally unrelated to the content - i would squash it with the second one to avoid confusion. 

Looks good to me otherwise, just a few questions.

File: libs/cardano-ledger-conformance/src/Test/Cardano/Ledger/Conformance/ExecSpecRule/Conway/Base.hs (hunk 1 above)
Comment (Soupstraw) [reply libs/cardano-ledger-conformance/src/Test/Cardano/Ledger/Conformance/ExecSpecRule/Conway/Base.hs:315]: Basically this function should compute `common` when given the other sets. It was completely wrong before, but it only matters when we want to check if some value conforms to the constraint so it's not awfully important for conformance testing where we only care about generating values.

File: eras/conway/impl/src/Cardano/Ledger/Conway/Rules/Ratify.hs (hunk 3 above)
Comment (Soupstraw) [reply eras/conway/impl/src/Cardano/Ledger/Conway/Rules/Ratify.hs:23]: Yup, thanks for reminding

**Review by lehins — APPROVED** [review]
Looks good. Needs a manual rebase, since there are some conflicts (likely due to SRP)
